server/src/main/java/org/elasticsearch/index/codec/PerFieldFormatSupplier.java

```diff
@@ -126,13 +146,7 @@ boolean useTSDBDocValuesFormat(final String field) {
     }
 
     private boolean excludeFields(String fieldName) {
-        // TODO: should we just allow all fields to use tsdb doc values codec?
-        // Avoid using tsdb codec for fields like _seq_no, _primary_term.
-        // But _tsid and _ts_routing_hash should always use the tsdb codec.
-        return fieldName.startsWith("_")
-            && fieldName.equals("_tsid") == false
-            && fieldName.equals("_ts_routing_hash") == false
-            && (SEQNO_FIELD_USE_TSDB_DOC_VALUES_FORMAT.isEnabled() && fieldName.equals(SeqNoFieldMapper.NAME) == false);
```

Comment:
Checking whether feature flag is enabled and whether seqno should not be excluded can't be expressed this way. Because if feature flag is disabled this then results in the field not being excluded.